### ISSUE TITLE
Document "CoapClient was reset" promise rejection

### ISSUE THREAD

**Issue by AlCalzone** [AlCalzone/node-tradfri-client#46]
see https://github.com/AlCalzone/node-tradfri-client/issues/44#issuecomment-372116455

**Comment by AlCalzone** [comment]
This can happen on any request/observation that has not completed yet when the user calls `reset()`. It's not a critical error but has to be there to notify the user that his request cannot be completed.

**Comment by neophob** [comment]
for reference, this error I captured:

```
Error: CoapClient was reset
    at normalizeReason (/myproject/node_modules/node-coap-client/build/lib/DeferredPromise.js:5:16)
    at Promise.rej [as reject] (/myproject/node_modules/node-coap-client/build/lib/DeferredPromise.js:13:36)
    at Function.reset (/myproject/node_modules/node-coap-client/build/CoapClient.js:191:33)
    at TradfriClient.reset (/myproject/node_modules/node-tradfri-client/build/tradfri-client.js:158:39)
    at tradfriClient.ping.catch.then (/myproject/app/lib/tradfriService.js:167:32)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
```

**Comment by neophob** [comment]
Another option would capture this rejected promise and use the error callback to signal an error

**Comment by AlCalzone** [comment]
The error pops up, because you call `reset` in your app:
> ```
> at TradfriClient.reset (/myproject/node_modules/node-tradfri-client/build/tradfri-client.js:158:39)
> at tradfriClient.ping.catch.then (/myproject/app/lib/tradfriService.js:167:32)
> ```

**Comment by neophob** [comment]
I know, it was just an idea, either update the docs that reset is an async function that returns a promise **OR** don't change the docs but use the error callback to signal the error.

**Comment by AlCalzone** [comment]
I'm leaning towards emitting a general library error and swallow the rejections. This means the corresponding documentation will have to state that all pending promises will neither be resolved or rejected anymore. Do you think this is good from a user's point of view?

Regarding your code snippet:
Are you catching rejections from `ping`? IMO that should always resolve and never reject. Or am I misunderstanding your stack frame?

**Comment by neophob** [comment]
I'm fine with swallowing the error and emitting an error - i think as long as the emitted error is understandable it should be fine and makes the implementation easier.

Yes I catch ping call, there was a rejection when I rebooted the gateway.

**Comment by AlCalzone** [comment]
> Yes I catch ping call, there was a rejection when I rebooted the gateway.

DTLS timeout? If so, that's a bug since `ping` should handle the errors itself.

**Comment by neophob** [comment]
I could not reproduce that error anymore with the `ping` function.

**Comment by AlCalzone** [comment]
"CoapClient was reset" should now be swallowed and transformed into an `"error"` event with a slightly more descriptive message.

**Comment by AlCalzone** [comment]
Released in v0.10.0
